clang/test/SemaCXX/constexpr-function-recovery-crash.cpp

```diff
@@ -99,6 +99,10 @@ TEST_EVALUATE(DoWhileCond, do {} while (some_cond < 10););    // expected-error
                                                               // expected-error {{constexpr variable 'forceEvaluateDoWhileCond' must be initialized by a constant expression}}
 TEST_EVALUATE(If, if (!!){};);                // expected-error + {{}}
 TEST_EVALUATE(IfInit, if (auto x = !!; 1){};);// expected-error + {{}}
-TEST_EVALUATE(ForInit, if (!!;;){};);         // expected-error + {{}}
```

Comment:
I think this case was use to check `for` statement but not `if`? Apologies if there is any mistake in my understanding.

Reply:
oops, yes, the intent was to check For here